### DISCUSSION TITLE
Share Assets - why not leverage /linkshare?

### DISCUSSION THREAD

**Discussion by Collingr** [Adobe-Marketing-Cloud/asset-share-commons#1018] [Ideas]
Hi all,
When sharing assets, recipients are hitting an SSO wall.
Managing a public and a private ASC portal is cumbersome.

Could we not leverage the JWT based authentication to access links leveraging the OOTB Linkshare that AEM Assets provides?

**Comment by tamelang** [comment]
A mechanism to pre sign urls for external sharing would be nice

**Reply by davidjgonzalez** [reply]
The OOTB Link share mechanics are pretty involved and dont have good public APIs; it would be alot of reverse engineering and hoping nothing changes that breaks it. I also dont recall if link share works on AEM Publish (which is where you should be sharing links from). 

I think a presigned URL would be the way to do this, but then we have to figure out what user the URL would sign you in as. From a practical POV, i think youd have to use whatever the intial auth token is (presigned URL, jwt, whatever) and trade it or a Cookie, since youd need all further interactions (ie HTTP requests) to be made "as that user".

Given AEM's authorizable/ACL paradgims, i think this is quite tricky.
